### PR TITLE
Update ViT loss compare in project/

### PR DESCRIPTION
**Pull request by rentainhe**
## TODO
- [x] 添加ViT的测试Loss对齐相关代码
- [x] 跑通Loss对齐
- [x] 添加自动下载数据集以及权重并对齐画图
- [x] 对齐Loss结果

## Loss对齐结果 - CIFAR10
误差在0.001之内
of_torch_diff
of_torch_loss

### CODE REVIEW

File: projects/vit_loss_compare/train_net.py [hunk 1]
```diff
@@ -13,28 +13,78 @@
...
+    for key, value in parameters.items():
+        if "num_batches_tracked" not in key:
+          val = value.detach().cpu().numpy()
+          # to global tensor
+          key, val = filter_keys(key, val)
+          val = flow.tensor(val).to_global(sbp=flow.sbp.broadcast, placement=flow.placement("cuda", {0: range(1)}))
+          new_parameters[key] = val
+    model.load_state_dict(new_parameters)
+    print("successfully load pytorch weight")
+    return model
```
Comment (L1aoXingyu) [inline projects/vit_loss_compare/train_net.py:68]: 这个部分可以考虑写到 utils 里面，把里面的 print 换成 logger

File: projects/vit_loss_compare/train_net.py [hunk 2]
```diff
@@ -13,28 +13,78 @@
...
 sys.path.append(".")
 
+import torch
+
+import oneflow as flow
+
 from libai.config import LazyConfig, default_argument_parser, try_get_key
 from libai.trainer import DefaultTrainer, default_setup
 from libai.utils.checkpoint import Checkpointer
+# from libai.models.vit_libai import VisionTransformer
```
Comment (CPFLAME) [inline projects/vit_loss_compare/train_net.py:28]: 这个注释是不是可以删了

Comment (rentainhe) [reply projects/vit_loss_compare/train_net.py:28]: 对，我删一下

File: projects/vit_loss_compare/train_net.py (hunk 1 above)
Comment (rentainhe) [reply projects/vit_loss_compare/train_net.py:68]: 好，我来改一下

File: loss_compare.sh [hunk 3]
```diff
@@ -1,7 +1,14 @@
 
-# bert loss compare
-python projects/bert_loss_compare/train_net.py --config-file projects/bert_loss_compare/configs/compare_loss.py 
+# # bert loss compare
+# python projects/bert_loss_compare/train_net.py --config-file projects/bert_loss_compare/configs/compare_loss.py 
+
+# # draw loss curve
+# echo "draw bert loss curve"
```
Comment (L1aoXingyu) [inline loss_compare.sh:6]: 这个打开一下

File: projects/vit_loss_compare/configs/compare_loss.py [hunk 4]
```diff
@@ -40,7 +42,7 @@
 )
 dataloader.train.dataset[0].transform = no_augmentation_transform
 
-dataloader.train.dataset[0].root = "/dataset/imagenet/extract"
+dataloader.train.dataset[0].root = "/dataset/extract"
```
Comment (L1aoXingyu) [inline projects/vit_loss_compare/configs/compare_loss.py:45]: 能不能用 cifar10 去做对齐，主要是因为后续可能采用数据集和权重自动下载的方式，imagenet 数据集太大了

File: loss_compare.sh (hunk 3 above)
Comment (rentainhe) [reply loss_compare.sh:6]: 哦哦忘记了

File: projects/vit_loss_compare/configs/compare_loss.py (hunk 4 above)
Comment (rentainhe) [reply projects/vit_loss_compare/configs/compare_loss.py:45]: 可以的，那我把Loss对齐换成CIFAR100上

**Review by L1aoXingyu — APPROVED** [review]
